File: k45/test_samples/bad_code.py
import os
import sys
import json
import time
import re

GLOBAL_CONFIG = {}
GLOBAL_CACHE = {}
GLOBAL_COUNTER = 0
GLOBAL_LOGGER = None

def long_function_with_many_params(a, b, c, d, e, f, g, h):
    global GLOBAL_COUNTER
    result = 0
    for i in range(100):
        result += i
        result += a
        result += b
        result += c
        result += d
        result += e
        result += f
        result += g
        result += h
        if result % 2 == 0:
            result = result * 2
        else:
            result = result * 3
        print(f"Current result: {result}")
        time.sleep(0.01)
        GLOBAL_COUNTER += 1
        if GLOBAL_COUNTER % 10 == 0:
            print("Counter reached multiple of 10")
    data = []
    for i in range(50):
        data.append(i * 2)
    processed = []
    for item in data:
        processed.append(item + 10)
    final = sum(processed)
    print(f"Final sum: {final}")
    return result

def another_duplicate_function(a, b, c, d, e, f, g, h):
    global GLOBAL_COUNTER
    result = 0
    for i in range(100):
        result += i
        result += a
        result += b
        result += c
        result += d
        result += e
        result += f
        result += g
        result += h
        if result % 2 == 0:
            result = result * 2
        else:
            result = result * 3
        print(f"Current result: {result}")
        time.sleep(0.01)
        GLOBAL_COUNTER += 1
        if GLOBAL_COUNTER % 10 == 0:
            print("Counter reached multiple of 10")
    data = []
    for i in range(50):
        data.append(i * 2)
    processed = []
    for item in data:
        processed.append(item + 10)
    final = sum(processed)
    print(f"Final sum: {final}")
    return result

class VeryLargeClass:
    def __init__(self):
        self.data1 = []
        self.data2 = {}
        self.data3 = set()
        self.config1 = None
        self.config2 = None
        self.config3 = None
    
    def method1(self):
        pass
    
    def method2(self):
        pass
    
    def method3(self):
        pass
    
    def method4(self):
        pass
    
    def method5(self):
        pass
    
    def method6(self):
        pass
    
    def method7(self):
        pass
    
    def method8(self):
        pass
    
    def method9(self):
        pass
    
    def method10(self):
        pass
    
    def method11(self):
        pass
    
    def method12(self):
        pass
    
    def method13(self):
        pass
    
    def method14(self):
        pass
    
    def method15(self):
        pass
    
    def method16(self):
        pass

def process_user_data(name, email, age, address, phone, city, country):
    return f"{name} - {email} - {age} - {address} - {phone} - {city} - {country}"

def save_user_data(name, email, age, address, phone, city, country):
    data = f"{name},{email},{age},{address},{phone},{city},{country}"
    with open('user.txt', 'w') as f:
        f.write(data)

def validate_user_data(name, email, age, address, phone, city, country):
    if not name or not email or not age:
        return False
    return True

File: k45/test_samples/bad_code.js
var globalCounter = 0;
var globalCache = {};
var globalConfig = {};

function longFunctionWithManyParams(a, b, c, d, e, f, g, h) {
    var result = 0;
    for (var i = 0; i < 100; i++) {
        result += i;
        result += a;
        result += b;
        result += c;
        result += d;
        result += e;
        result += f;
        result += g;
        result += h;
        if (result % 2 === 0) {
            result = result * 2;
        } else {
            result = result * 3;
        }
        console.log("Current result: " + result);
        globalCounter++;
        if (globalCounter % 10 === 0) {
            console.log("Counter reached multiple of 10");
        }
    }
    var data = [];
    for (var j = 0; j < 50; j++) {
        data.push(j * 2);
    }
    var processed = [];
    for (var k = 0; k < data.length; k++) {
        processed.push(data[k] + 10);
    }
    var final = processed.reduce(function(sum, val) { return sum + val; }, 0);
    console.log("Final sum: " + final);
    return result;
}

function anotherDuplicateFunction(a, b, c, d, e, f, g, h) {
    var result = 0;
    for (var i = 0; i < 100; i++) {
        result += i;
        result += a;
        result += b;
        result += c;
        result += d;
        result += e;
        result += f;
        result += g;
        result += h;
        if (result % 2 === 0) {
            result = result * 2;
        } else {
            result = result * 3;
        }
        console.log("Current result: " + result);
        globalCounter++;
        if (globalCounter % 10 === 0) {
            console.log("Counter reached multiple of 10");
        }
    }
    var data = [];
    for (var j = 0; j < 50; j++) {
        data.push(j * 2);
    }
    var processed = [];
    for (var k = 0; k < data.length; k++) {
        processed.push(data[k] + 10);
    }
    var final = processed.reduce(function(sum, val) { return sum + val; }, 0);
    console.log("Final sum: " + final);
    return result;
}

class VeryLargeClass {
    constructor() {
        this.data1 = [];
        this.data2 = {};
        this.data3 = new Set();
        this.config1 = null;
        this.config2 = null;
        this.config3 = null;
    }
    
    method1() {}
    method2() {}
    method3() {}
    method4() {}
    method5() {}
    method6() {}
    method7() {}
    method8() {}
    method9() {}
    method10() {}
    method11() {}
    method12() {}
    method13() {}
    method14() {}
    method15() {}
    method16() {}
}

function processUserData(name, email, age, address, phone, city, country) {
    return name + " - " + email + " - " + age + " - " + address;
}

function saveUserData(name, email, age, address, phone, city, country) {
    var data = name + "," + email + "," + age + "," + address;
    localStorage.setItem('user', data);
}

function validateUserData(name, email, age, address, phone, city, country) {
    if (!name || !email || !age) {
        return false;
    }
    return true;
}
